reporting/performance/test_mode.cpp

```diff
@@ -0,0 +1,126 @@
+//  (C) Copyright Nick Thompson and Matt Borland 2020.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <random>
+#include <boost/math/statistics/univariate_statistics.hpp>
+#include <benchmark/benchmark.h>
+
+template <class Z>
+void test_mode(benchmark::State& state)
+{
+    using boost::math::statistics::mode;
+    
+    std::random_device rd;
+    std::mt19937_64 mt(rd());
+    std::uniform_real_distribution<double> dist {1, 10};
+
+    auto gen = [&dist, &mt](){return dist(mt);};
+
+    std::vector<Z> v(state.range(0));
+    std::generate(v.begin(), v.end(), gen);
+
+    for (auto _ : state)
+    {
+        benchmark::DoNotOptimize(mode(v));
```

Comment:
I would only  benchmark `sorted_mode`, since the performance of `std::sort` is understood.